plugins/performance-lab/includes/site-health/effective-asset-cache-headers/hooks.php

```diff
@@ -26,6 +26,16 @@ function perflab_effective_asset_cache_headers_add_test( array $tests ): array {
 		'label' => __( 'Effective Caching Headers', 'performance-lab' ),
 		'test'  => 'perflab_effective_asset_cache_headers_assets_test',
 	);
+
+	/**
+	 * Static assets are expected to not have effective cache headers in non-production environments.
+	 *
+	 * GH Issue: https://github.com/WordPress/performance/issues/2031
+	 */
+	if ( in_array( wp_get_environment_type(), array( 'local', 'development' ), true ) ) {
+		unset( $tests['direct']['effective_asset_cache_headers'] );
+	}
```

Comment:
Instead of unsetting something that was set, why not just short-circuit the function to avoid setting it in the first place, or rather wrap the setting with the `if` statement, so the entire function body can be:

```php
if ( ! in_array( wp_get_environment_type(), array( 'local', 'development' ), true ) ) {
	$tests['direct']['effective_asset_cache_headers'] = array(
		'label' => __( 'Effective Caching Headers', 'performance-lab' ),
		'test'  => 'perflab_effective_asset_cache_headers_assets_test',
	);
}
return $tests;
```